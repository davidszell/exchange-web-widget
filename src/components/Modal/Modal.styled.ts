import styled from "styled-components";

export const Container = styled.div`
    position: fixed;
    left: 0;
    right: 0;
    top: 0;
    bottom: 0;
    background-color: rgba(0, 0, 0, 0.2);
    display: flex;
    align-items: center;
    justify-content: center;
`;

export const Content = styled.div`
    width: 500px;
    max-height: 500px;
    overflow-y: auto;
    background-color: white;
`;

export const CloseButton = styled.button`
    
`;