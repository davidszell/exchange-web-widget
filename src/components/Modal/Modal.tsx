import React from 'react';
import { CloseButton, Container, Content } from './Modal.styled';

type ModalPropsType = {
  children: React.ReactNode,
  handleClose: () => void
};

const Modal = ({ children, handleClose }: ModalPropsType): JSX.Element => {
  const handleCloseClick = () => {
    handleClose();
  };

  return (
    <Container data-testid="modal">
      <Content>
        <CloseButton onClick={handleCloseClick}>X</CloseButton>
        { children }
      </Content>
    </Container>
  );
};

export default Modal;
