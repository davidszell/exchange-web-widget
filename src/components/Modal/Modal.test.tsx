import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import Modal from './Modal';

describe('Modal', () => {

    it('matches snapshot', () => {
        const tree = renderer
            .create(<Modal handleClose={jest.fn()}>Content</Modal>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    })
})