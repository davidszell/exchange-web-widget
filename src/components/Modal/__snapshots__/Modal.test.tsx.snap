// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal matches snapshot 1`] = `
.c2 {
  border: 1px solid #157ae2;
  color: #157ae2;
  background-color: white;
  padding: 10px;
  border-radius: 5px;
  cursor: pointer;
  font-weight: 500;
}

.c2:hover {
  color: white;
  border-color: #054A92;
  background-color: #054A92;
}

.c2:disabled,
.c2[disabled],
.c2:disabled:hover,
.c2[disabled]:hover {
  cursor: unset;
  color: #777;
  border-color: #ddd;
  background-color: #ddd;
}

.c0 {
  position: fixed;
  left: 0;
  right: 0;
  top: 0;
  bottom: 0;
  background-color: rgba(0,0,0,0.2);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  width: 500px;
  max-height: 500px;
  overflow-y: auto;
  background-color: white;
  padding: 10px;
  border-radius: 10px;
  position: relative;
}

.c3 {
  position: absolute;
  right: 10px;
  top: 10px;
}

<div
  className="c0"
  data-testid="modal"
>
  <div
    className="c1"
  >
    <button
      className="c2 c3"
      onClick={[Function]}
    >
      X
    </button>
    Content
  </div>
</div>
`;
